src/window.py

```diff
@@ -281,6 +284,23 @@ def _move(self, path, navigating=False):
         else:
             self._open(path)
 
+
+            self._places.add_starred(path)
+
+        self._path = path
+        self.bookmark_button.props.sensitive = self._places.canBookmark(path)
+        self._update_bookmark_icon()
+
+    def _update_bookmark_icon(self):
```

Comment:
This could be moved to a custom BookmarkButton widget, and handle the logic there. It would be specially good if the CSS trick works and requires adding and removing CSS classes to the widget. Then the widget would simply expose an "update" method or something for use cases like this.